src/main/java/org/opensearch/security/securityconf/impl/SecurityDynamicConfiguration.java

```diff
@@ -311,6 +325,18 @@ public Class<?> getImplementingClass() {
     }
 
     @SuppressWarnings("unchecked")
+    @JsonIgnore
+    public SecurityDynamicConfiguration<T> clone() {
```

Comment:
why create a shallow copy ?

Reply:
The clone is not completely shallow - it also includes a copy of the `centries` map. The only thing not cloned are the actual `RoleV7` instances (or related).

It is used here in order to be able to add the static roles without side effects:

https://github.com/opensearch-project/security/blob/1fef11be731e5e2a3c06c5d565faf23525ffe0d6/src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java#L221-L244

The deep clone would be just a waste of CPU cycles here.

Reply:
Wouldn't `centries` still point to same memory location since we are adding references? Or maybe I'm reading this wrong.

Reply:
The clone() method creates a copy of the `centries` `HashMap`:

https://github.com/opensearch-project/security/blob/1fef11be731e5e2a3c06c5d565faf23525ffe0d6/src/main/java/org/opensearch/security/securityconf/impl/SecurityDynamicConfiguration.java#L329-L338